components/navigation/DesktopNav.tsx

```diff
@@ -95,7 +92,7 @@ const DesktopNav = () => {
     return () => {
       supabase.removeChannel(channel);
     };
-  }, [trigger, pathname, userId]);
+  }, [notification, pathname, userId]);
```

Comment:
I didn't spot this before but please adjust this name to reflect a boolean - `hasNotifications` or something